mlflow/utils/rest_utils.py

```diff
@@ -41,15 +32,17 @@ def databricks_api_request(endpoint, method, req_body_json=None, params=None):
         headers = None
     # TODO: Remove `verify=False`, currently need it to run against dev shards.
     return http_request(hostname=hostname, endpoint=final_endpoint, method=method, auth=auth,
-                        headers=headers, req_body_json=req_body_json, params=params)
+                        headers=headers, req_body_json=req_body_json, params=params,
+                        verify_success=verify_success)
 
 
-def http_request(hostname, endpoint, method, auth, headers, req_body_json, params, retries=3):
+def http_request(hostname, endpoint, method, auth, headers, req_body_json, params,
+                 retries=3, verify_success=True):
     url = "%s%s" % (hostname, endpoint)
     for i in range(retries):
         response = requests.request(method=method, url=url, headers=headers, verify=False,
                                     params=params, json=req_body_json, auth=auth)
-        if response.status_code == 200:
+        if not verify_success or response.status_code == 200:
             return json.loads(response.text)
         else:
             eprint("API request to %s failed with code %s != 200, retrying up to %s more times. "
```

Comment:
Btw, I'm not sure we should retry on 400-level error codes, as these are application level responses (as opposed to things like timeouts [-1], or transient/system errors [500-level]).

For example, an invalid cluster spec returns a lot of output, especially due to retries:

```
API request to https://aaf2364b4896d11e8b8d202c4bfdb663-840219565.us-west-2.elb.amazonaws.com/api/2.0/jobs/runs/submit failed with code 400 != 200, retrying up to 1 more times. API response body: {"error_code":"INVALID_PARAMETER_VALUE","message":"Missing required field: settings.cluster_spec.new_cluster.size"}
```